src/imagej/dims.py

```diff
@@ -299,19 +284,56 @@ def _get_enumerated_axis():
     return sj.jimport("net.imagej.axis.EnumeratedAxis")
 
 
-def _get_linear_axis(axis_type: "jc.AxisType", values):
+def _get_linear_axis(axis_type: "jc.AxisType", scale_array: np.ndarray, values):
     """Get linear axis.
 
     This is used if no EnumeratedAxis is found. If EnumeratedAxis
     is available, use _get_enumerated_axis() instead.
     """
     DefaultLinearAxis = sj.jimport("net.imagej.axis.DefaultLinearAxis")
     origin = values[0]
-    scale = values[1] - values[0]
+    if len(scale_array) <= 1:
+        scale = 1
+    else:
+        scale = _compute_slope(scale_array)
     axis = DefaultLinearAxis(axis_type, scale, origin)
     return axis
 
 
+def _compute_slope(axis):
+    """
+    Compute slope (i.e. scale) of an axis with all points.
+
```

Comment:
I can't say I really understand what this means. What slope is calculated? Can we elaborate a bit in this docstring?

Reply:
Yes, I agree this is nebulous. I'll improve this docstring. But to briefly summarize what's going on here:
 
Axes have coordinates along each axis (a 1D array) which can be linear or non-linear. When a user converts an `xarray.DataArray` with coordinates. Newer versions of ImageJ come with `EnumeratedAxis` which makes converting the coordinates really easy and also handles non-linear coordinates. If `EnumeratedAxis` is not available however, we fall back to `DefaultLinearAxis` which requires the slope of the 1D coordinate array. Prior to this PR we calculated the slope by subtracting the first two elements from each other from the coordinate array. I originally had implemented linear regression so we could actually decide if coordinates were linear or not. But that was kind of silly and not really necessary. Part of that however, involved calculating the slope using all points of the array instead of just the first two elements. I thought this was a more robust slope calculation and got values that matched ImageJ's slope values (`float`).

Reply:
Gotcha, thanks for the explanation. Having this explanation in the docstring would be great!

Reply:
Improved the docstring: https://github.com/imagej/pyimagej/pull/238/commits/0126d0039cc2a6bc7b5531bce19a2c2e28825790